sceptre-example/config/lambda-stack-bootstrapped.yaml

```diff
@@ -0,0 +1,73 @@
+# This is an example of deploying the CDK using the "bootstrapped" deployment_type, which means
+# using the resources from the CDK boostrap stack for things such as the asset bucket, asset ECR
+# repo, as well as the roles used to push those assets to the cloud. You can use either a Bootstrap
+# stack deployed via Sceptre or an existing one deployed to your account via CDK. All you need to
+# do to reference a given bootstrap stack in that environment is to use the bootstrap stack's
+# qualifier as the "bootstrap_qualifier". If you don't use a bootstrap_qualifier, CDK will default
+# to the default qualifier and try to use a bootstrap stack deployed in your account that has that
+# qualifier.
+#
+# Regardless of how the bootstrap stack is deployed, it should be generated by `cdk bootstrap
+# --show-template` to ensure maximum compatibility.
+#
+# IMPORTANT: Even when using the "bootstrapped" deployment_type, Sceptre WILL NOT use the actual
+# bootstrapped deployment role to deploy the CloudFormation stack; It will use its own configured
+# deployment environment, profile, and/or iam_role.
+template:
+  type: cdk
+  # The path is always within your project's templates/ directory.
+  path: lambda_stack.py
+  deployment_type: bootstrapped
```

Comment:
what happens if a user deploys multiple `boostrapped` templates?  Would things only get created on the 1st boostrapped deployment and then it's noop after that?

Reply:
Nope. Using the bootstrapped template means we use the conventional CDK stack synthesizer. The conventional stack synthesizer will synthesize the template and then we'll push the assets to the cloud using the bootstrap stack. CDK finds the appropriate bootstrapped resources to push those assets using the bootstrap qualifier.

Reply:
I assumed that a `boostrap` only needs to be done once?  Why would you need to boostrap multiple times and why would you need multiple boostrap assets per aws account?  or is it that CDK was just designed to boostrap multiple assets for some reason?

Reply:
CDK is perfectly compatible with having multiple bootstrap stacks in an AWS account. That's the point of using the qualifier; By specifying the qualifier, you can tell CDK which bootstrap stack to use. This is how my own organization uses CDK. We deploy one bootstrap stack per _project_. Each bootstrap stack has its own list of specially configured permissions on what can and cannot be done via automated deployment as well as permissions boundaries that ensure the project cannot self-escalate its own permissions or touch highly protected resources in our infrastructure. We do it this way to ensure the Principle of Least Privilege is followed in our infrastructure. We're a fin-tech, so there are special concerns around protecting cardholder data and PII (and the application infrastructure based around those). Each bootstrap stack (with its policies/boundaries) is reviewed explicitly by our InfoSec department and then explicitly deployed by one of our admins. But once we have those in place, we can set up automated infrastructure deployments that don't need explicit approval from an admin for those projects. 

CDK has a default qualifier, so if you don't pay any attention to the qualifier stuff, it defaults to having one bootstrap stack per account and CDK will just look for that default qualifier. But obviously, there are use cases beyond that simplest one (such as my own org's), so I wanted to make sure this handler was fully compatible with all those use cases.

Reply:
thanks for the explanation